### ISSUE TITLE
Report linting warnings to PR

### ISSUE THREAD

**Issue by cojoj** [JohnSundell/Marathon#102]
Our CI is printing linter errors/warnings just fine, but the results aren't visible to the PR creators. - which kinda sucks, right? 😜
Do you know how to achieve this @pixyzehn?
One thing that comes to my mind is to introduce [Danger](https://github.com/danger/danger) ([since buddybuild uses it without any issues](https://github.com/danger/danger/issues/695)) and use something like [this](https://github.com/ashfurrow/danger-swiftlint).

**Comment by pixyzehn** [comment]
@cojoj You can see the errors/warnings in buddybuild logs for now like this.
https://dashboard.buddybuild.com/public/apps/58ff19a79a06210001d14c2d/build/5937a686f8735a0001c4b64b#logs

But I know it's a bit troublesome... if we put a high priority on fixing the errors/warnings by SwiftLint, it's better to use such a tool.
The one bad thing is that we have to maintenance such a dependency.
@JohnSundell What do you think?

**Comment by cojoj** [comment]
Hah, dependencies... Dependencies everywhere... ¯\_(ツ)_/¯

**Comment by pixyzehn** [comment]
Yes, but I understand Danger helps us a lot especially for a big OSS. Maybe it's time to introduce it!

**Comment by cojoj** [comment]
It's a nice addition for keeping consistency in project. Especially when it comes to modifying more than just a source files. Right now you have to remember everything by heart... If you add something, update README etc.

**Comment by JohnSundell** [comment]
Sounds great! I'd love to add Danger to Marathon. I'll update the issue title to "Add Danger" and set it to ready for implementation. Anyone up for the task? 😄

**Comment by cojoj** [comment]
I can play with it @JohnSundell. I always wanted to give Danger a spin 😎

**Comment by JohnSundell** [comment]
@cojoj Awesome 🚀

**Comment by cojoj** [comment]
I'm working on it in #106 🙌 Feel free to write what you;d like to see in `Dangefile`

**Comment by cojoj** [comment]
It's super strange that I can't make SwiftLint work with our CI... It's not reporting at all! I don't know what's the issue... I've tried looking up some similar _set ups_ and for wxample [firefox](https://github.com/mozilla-mobile/firefox-ios) has it and it works just fine. @pixyzehn have you ever faced such issue? I don't know if it's a bug or maybe I'm missing something in integration...

**Comment by pixyzehn** [comment]
@cojoj I have never faced that...🤔

**Comment by cojoj** [comment]
@pixyzehn FYI solved it... Looks like our `.swiftlint.yml` had _some_ conflicts. If you're interested you can take a look at: https://github.com/ashfurrow/danger-swiftlint/issues/43

Also, please take a look and verify `.swiftlint.yml` in #106
